src/posts/posts.controller.ts

```diff
@@ -62,7 +63,8 @@ export class PostsController {
   }
 
   @Delete('/:postid')
-  deletePost() {
-    throw new NotImplementedException();
+  @HttpCode(204)
+  deletePost(@Param('postid') postid: string) {
+    return this.postsService.deletePost(postid);
```

Comment:
You must return in the controller in Nestjs for the endpoint to work correctly. Not returning will result in the endpoint always responding with the default status code, which is `200` by default if not stated otherwise. Here it will always respond with `204`.

Reply:
Yes If we remove the `return` keyword, the response will be `204` no matter what. However, a fix for this by adding `async` `await` without using `return`. In such case, nest will not respond with `204` unless it is successfully deleted.

Reply:
I find this misleading, more of a workaround than a fix. `return`ing `void` is not something to be avoided in the first place to search for a workaround to it.

Reply:
I don't think it is misleading. We use `async` `await` as we are actually waiting for `service.deletePost()` to finish its job before continuing with the rest of the code in our method. So, semantically it is correct to use it here. It also will fix the `return` issue, instead of returning void from the method.